features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Scopes/Scope.js

```diff
@@ -0,0 +1,138 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react'
+import { Input, Icon, Checkbox, Button, Card, Tag, Form } from 'antd';
+import { Row, Col } from 'antd';
+import Api from "../../../../data/api";
+import {message} from "antd/lib/index";
+import TagsInput from 'react-tagsinput'
+import 'react-tagsinput/react-tagsinput.css'
+import Loading from '../../../Base/Loading/Loading'
+
+class Scope extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            visible: false,
+            tags: []
+        };
+        this.toggleScopeData = this.toggleScopeData.bind(this);
+        this.removeScope = this.removeScope.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChangeDescription = this.handleChangeDescription.bind(this);
+    }
+    toggleScopeData(){
+        const api = new Api();
+        let promised_scopes_object = api.getScopeDetail(this.props.api_uuid,this.props.name);
+        promised_scopes_object.then(
+            response => {
+                this.setState({apiScope: response.obj});
+            }
+        ).catch(
+            error => {
+                if (process.env.NODE_ENV !== "production") {
+                    console.log(error);
+                }
+                let status = error.status;
+                if (status === 404) {
+                    this.setState({notFound: true});
+                }
+            }
+        );
+        this.setState({visible: !this.state.visible});
+    }
+    removeScope() {
+        let scope_name = this.props.name;
+        const hideMessage = message.loading("Deleting the Scope ...", 0);
+        const api = new Api();
+        let promised_scope_delete = api.deleteScope(this.props.api_uuid, scope_name);
+        promised_scope_delete.then(
+            response => {
+                if (response.status !== 200) {
+                    console.log(response);
+                    message.error("Something went wrong while deleting the " + scope_name + " Scope!");
+                    hideMessage();
+                    return;
+                }
+                message.success(scope_name + " Scope deleted successfully!");
+                this.props.deleteScope(scope_name);
+                hideMessage();
+            }
+        );
+    }
+    handleChange(roles) {
+        let apiScope = this.state.apiScope;
+        apiScope.bindings.values = roles;
+        this.setState({apiScope:apiScope})
+    }
+    handleChangeDescription(e){
+        let apiScope = this.state.apiScope;
```

Comment:
we could use [Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
```javascript
 const {apiScope} = this.state;
``` 
so no repeating names